explorer-ui/src/components/LoginForm/RegisterForm/RegisterForm.jsx

```diff
@@ -0,0 +1,168 @@
+import * as React from "react"
+import "./RegisterForm.css"
+import axios from "axios"
+import * as config from "../../../config"
+import styled from "styled-components";
+import { Link } from "react-router-dom";
+import Footer from "../../Home/Footer/Footer";
+import NavBar from "../../Home/NavBar/NavBar";
+export default function RegisterForm({ handleLogin }) {
+    const username = React.createRef();
+    const password = React.createRef();
+    const age = React.createRef();
+    const email = React.createRef();
+
+    const handleSubmit = event => {
+        event.preventDefault();
+
+        const register = async () => {
+            try {
+                const res = await axios.post(`${config.API_BASE_URL}/register`, {
+                    "username" : username.current.value,
+                    "password" : password.current.value,
+                    "age": age.current.value,
+                    "email": email.current.value
+                    })
+                handleLogin(res.data.user)    
+            } catch (err) {
+                alert(err)
+                console.log(err)
+            }
+        }
+        register()
+    }
```

Comment:
so the handleSubmit can be an async fxn itself instead of the nesting

then in the button or wherever its reference can use an anonymous fxn to call it i.e {async () => await handleSubmit()}